src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala

```diff
@@ -14,13 +14,94 @@ import scala.concurrent.duration._
+
+  case class TcpRequestMapping(port: Int) extends RequestMapping
+
+  trait ProtocolFamilyRequestMappings {
+    type Mapping <: RequestMapping
+    def protocolFamily: String
+    def requestMappings: Set[Mapping]
+  }
+
+  object HttpRequestMappings {
+    def apply(httpRequestMappings: HttpRequestMapping*): HttpRequestMappings =
+      HttpRequestMappings(httpRequestMappings.toSet)
+  }
+
+  case class HttpRequestMappings(requestMappings: Set[HttpRequestMapping]) extends ProtocolFamilyRequestMappings {
+    override type Mapping = HttpRequestMapping
+    val protocolFamily = "http"
+  }
+
+  object TcpRequestMappings {
+    def apply(ports: Int*): TcpRequestMappings =
+      TcpRequestMappings(ports.toSet.map(TcpRequestMapping))
+  }
+
+  case class TcpRequestMappings(requestMappings: Set[TcpRequestMapping]) extends ProtocolFamilyRequestMappings {
+    override type Mapping = TcpRequestMapping
+    val protocolFamily = "tcp"
+  }
+
+  object Acl {
+    def apply(protocolFamilyRequestMappings: ProtocolFamilyRequestMappings*): Acl =
+      Acl(protocolFamilyRequestMappings.toSet)
+  }
+
+  case class Acl(requestMappings: Set[ProtocolFamilyRequestMappings])
+
+  object Endpoint {
+    /**
+     * Represents a service endpoint.
+     * @param bindProtocol the protocol to bind for this endpoint, e.g. "http"
+     * @param bindPort the port the bundle component’s application or service actually binds to; when this is 0 it will be dynamically allocated (which is the default)
+     * @param services the public-facing ways to access the endpoint form the outside world with protocol, port, and/or path
+     */
+    @deprecated("services are deprecated in favour of ACLs", since = "1.2.2")
+    def apply(bindProtocol: String, bindPort: Int = 0, services: Set[URI] = Set.empty): Endpoint =
+      new Endpoint(bindProtocol, bindPort, Some(services), None)
+
+    def apply(bindProtocol: String, bindPort: Int, acls: Acl*): Endpoint =
+      new Endpoint(bindProtocol, bindPort, None, Some(acls.toSet))
+  }
+
   /**
    * Represents a service endpoint.
    * @param bindProtocol the protocol to bind for this endpoint, e.g. "http"
    * @param bindPort the port the bundle component’s application or service actually binds to; when this is 0 it will be dynamically allocated (which is the default)
-   * @param services the public-facing ways to access the endpoint form the outside world with protocol, port, and/or path
+   * @param services **deprecated** - the public-facing ways to access the endpoint form the outside world with protocol, port, and/or path
+   * @param acls - list of protocol and its corresponding paths exposed by the service endpoint
    */
-  case class Endpoint(bindProtocol: String, bindPort: Int = 0, services: Set[URI] = Set.empty)
+  case class Endpoint(bindProtocol: String, bindPort: Int,
+    @deprecated("services are deprecated in favour of ACLs", since = "1.2.2") services: Option[Set[URI]],
```

Comment:
I think we should make this version 1.3.0 as the feature of acls is reasonably significant.